src/views/Admin/AdminUsers.vue

```diff
@@ -0,0 +1,180 @@
+<template>
+  <div class="admin-users">
+    <div class="search-panel">
+      <div class="col">
+        <div class="row">
+          <label for="first-name" class="uc-form-label">First name</label>
+          <input id="first-name" type="text" v-model="firstName" />
+        </div>
+        <div class="row">
+          <label for="last-name" class="uc-form-label">Last name</label>
+          <input id="last-name" type="text" v-model="lastName" />
+        </div>
+        <div class="row">
+          <label for="email" class="uc-form-label">Email</label>
+          <input id="email" type="text" v-model="email" />
+        </div>
+      </div>
+
+      <div>
+        <button class="uc-form-button" type="button" @click="getUsers">
+          Search
+        </button>
+      </div>
+    </div>
+    <page-control
+      :page="page"
+      :isFirstPage="isFirstPage"
+      :isLastPage="isLastPage"
+      @nextPage="nextPage"
+      @previousPage="previousPage"
+    />
+    <div class="list-wrapper">
+      <div class="list">
+        <user-list-item v-for="user in users" :key="user._id" :user="user" />
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import NetworkService from "@/services/NetworkService";
+import PageControl from "@/components/Admin/PageControl";
+import UserListItem from "@/components/Admin/UserListItem";
+
+const getUsers = async data => {
+  const {
+    body: { users, isLastPage }
+  } = await NetworkService.adminGetUsers(data);
+
+  return { users, isLastPage };
+};
+
+export default {
+  name: "AdminUsers",
+
+  components: { UserListItem, PageControl },
+
+  data() {
+    return {
+      page: 1,
+      isLastPage: true,
+      users: [],
+      firstName: "",
+      lastName: "",
+      email: ""
+    };
+  },
+
+  async created() {
+    const {
+      query: { page: pageQuery, firstName, lastName, email }
+    } = this.$route;
+    const page = parseInt(pageQuery);
+    this.page = page || this.page;
+    this.firstName = firstName || this.firstName;
+    this.lastName = lastName || this.lastName;
+    this.email = email || this.email;
+
+    this.$nextTick(() => {
```

Comment:
is the purpose of `nextTick` here to wait for the dom to be accessible?

would putting this in the `mounted` lifecycle hook be equivalent? (don't need to change, just curious why you went this route)

Reply:
Yup to both of those questions! Alternatively, like you said, we can just remove `nextTick` and place the event listener in the `mounted` hook and it will work fine as well. I didn't have a real reason outside of copying it from another component and forgetting to update haha, but it's probably better to have the listeners fired from the `mounted` hook whenever they can be